packages/adblocker/src/data-view.ts

```diff
@@ -389,23 +391,17 @@ export class StaticDataView {
   }
 
   public pushUTF8(raw: string): void {
-    const str = encode(raw);
-    this.pushLength(str.length);
-
-    for (let i = 0; i < str.length; i += 1) {
-      this.buffer[this.pos++] = str.charCodeAt(i);
-    }
+    const { written } = TEXT_ENCODER.encodeInto(raw, this.buffer.subarray(this.pos + 4));
+    this.pushLength(written);
```

Comment:
Since you do `+4` here (32 bits), it's not useful to use `pushLength` since you might as well push the length always as uint32. The only reason to use pushLength is to use less bits to encode the length of small strings.

Reply:
Yup, I'm on the fix to support dynamic positioning to utilize `pushLength`.

Reply:
@remusao Please, review [`0339bdc` (#4513)](https://github.com/ghostery/adblocker/pull/4513/commits/0339bdca5b8e58405f00c9b051950297e6e2e8a2)

Reply:
Dropped useless fast path https://github.com/ghostery/adblocker/pull/4513/files#diff-02c5b76ebc841a4005b9541e8e09eb74aecf407dfd69e179c028c6c83a771552R93-R96